command/meta.go

```diff
@@ -176,7 +177,34 @@ func (m *Meta) allNamespaces() bool {
 }
 
 func (m *Meta) Colorize() *colorstring.Colorize {
-	_, coloredUi := m.Ui.(*cli.ColoredUi)
+	ui := m.Ui
+	coloredUi := false
+
+	// Meta.Ui may wrap other cli.Ui instances, so unwrap them until we find a
+	// *cli.ColoredUi or there is nothing left to unwrap.
+	for {
+		_, coloredUi = ui.(*cli.ColoredUi)
+		if coloredUi {
+			break
+		}
+
+		v := reflect.ValueOf(ui)
+		if v.Kind() == reflect.Ptr {
+			v = v.Elem()
+		}
+		for i := 0; i < v.NumField(); i++ {
```

Comment:
Missed this before I approved, but there's a test panic at this line:

```
=== FAIL: command TestMeta_Colorize (0.00s)
panic: reflect: call of reflect.Value.NumField on zero Value [recovered]
	panic: reflect: call of reflect.Value.NumField on zero Value

goroutine 51085 [running]:
testing.tRunner.func1.2({0x2905000, 0xc0088c53f8})
	/opt/hostedtoolcache/go/1.20.2/x64/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.20.2/x64/src/testing/testing.go:1529 +0x39f
panic({0x2905000, 0xc0088c53f8})
	/opt/hostedtoolcache/go/1.20.2/x64/src/runtime/panic.go:884 +0x213
reflect.flag.mustBe(...)
	/opt/hostedtoolcache/go/1.20.2/x64/src/reflect/value.go:223
reflect.Value.NumField({0x0?, 0x0?, 0x2e26140?})
	/opt/hostedtoolcache/go/1.20.2/x64/src/reflect/value.go:2041 +0x8d
github.com/hashicorp/nomad/command.(*Meta).Colorize(0x35aac08?)
	/home/runner/work/nomad/nomad/command/meta.go:195 +0x23b
github.com/hashicorp/nomad/command.TestMeta_Colorize.func8(0xc001930f60?)
	/home/runner/work/nomad/nomad/command/meta_test.go:155 +0x16b
testing.tRunner(0xc00df2a4e0, 0xc008a40920)
	/opt/hostedtoolcache/go/1.20.2/x64/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.20.2/x64/src/testing/testing.go:1629 +0x3ea
```

Reply:
Oof...`reflect` is scary 😅 

Fixed in https://github.com/hashicorp/nomad/pull/16278/commits/e53fbe00e4d0bc5d95fa25ee3cb4cf3ea8b1a768, I needed to check for `nil` at the top of the loop 🤦